src/helpers/amazon-q/amazon-q-helpers.ts

```diff
@@ -1,86 +1,31 @@
-import { SlackEventsEnv } from '@functions/slack-event-handler';
 import { createButton, getMarkdownBlocks, SLACK_ACTION } from '@helpers/slack/slack-helpers';
 import { makeLogger } from '@src/logging';
 import { isEmpty } from '@src/utils';
-import { ChatDependencies } from '@src/helpers/chat';
 import { Block } from '@slack/web-api';
-import { ChatSyncCommandOutput, AttachmentInput } from '@aws-sdk/client-qbusiness';
-import { Credentials } from 'aws-sdk';
-import { ExpiredTokenException } from '@aws-sdk/client-sts';
+import { ChatSyncCommandOutput, SourceAttribution, MetadataEvent } from '@aws-sdk/client-qbusiness';
 
 const logger = makeLogger('amazon-q-helpers');
 
-// Member must have length less than or equal to 7000
-const AMAZON_Q_MSG_LIMIT = 7000;
-const WARN_TRUNCATED = `| Please note that you do not have all the conversation history due to limitation`;
 
-export const chat = async (
-  slackUserId: string,
-  incomingMessage: string,
-  attachments: AttachmentInput[],
-  dependencies: ChatDependencies,
-  env: SlackEventsEnv,
-  iamSessionCreds: Credentials,
-  context?: {
-    conversationId: string;
-    parentMessageId: string;
-  }
-): Promise<ChatSyncCommandOutput | Error> => {
-  try {
-    // Enforce max input message limit - may cause undesired side effects
-    // TODO consider 'smarter' truncating of number of chat history messages, etc. rather
-    // than simple input string truncation which may corrupt JSON formatting of message history
-    const inputMessage =
-      incomingMessage.length > AMAZON_Q_MSG_LIMIT
-        ? incomingMessage.slice(
-            incomingMessage.length + WARN_TRUNCATED.length - AMAZON_Q_MSG_LIMIT
-          ) + WARN_TRUNCATED
-        : incomingMessage;
-
-    const response = await dependencies.callClient(
-      slackUserId,
-      inputMessage,
-      attachments,
-      env,
-      iamSessionCreds,
-      context
-    );
-    logger.debug(`AmazonQ chatSync response: ${JSON.stringify(response)}`);
-    return response;
-  } catch (error) {
-    logger.error(`Caught Exception: ${JSON.stringify(error)}`);
-    if (error instanceof Error) {
-      logger.debug(error.stack);
-      if (error instanceof ExpiredTokenException) {
-        logger.error(`Token expired: ${error.message}`);
-      }
-      return new Error(error.message);
-    } else {
-      return new Error(`${JSON.stringify(error)}`);
-    }
-  }
-};
-
-export const getResponseAsBlocks = (response: ChatSyncCommandOutput) => {
-  if (isEmpty(response.systemMessage)) {
+export const getResponseAsBlocks = (content: string, systemMessageId: string, sourceAttributions: SourceAttribution[]) => {
```

Comment:
what was the reason for changing the parameter from 1 response to its contents here? I think the former is more readable

Reply:
This is a nit, don't need to block on this

Reply:
Ack. Same as `getFeedbackBlocks`, output is different based on ChatSync/Chat but they both use the same helpers. MetadataEvent has all the same attributes as a ChatSync response, but when updating the slack chat we aren't guaranteed to have a MetadataEvent by the time the first buffer fills -> only have access to latestTextEvent which doesn't have sourceAttributions. I could maybe rewrite with a single response but then there's conditionals in the input.